roofit/multiprocess/src/HeatmapAnalyzer.cxx

```diff
@@ -0,0 +1,212 @@
+/*
+ * Project: RooFit
+ * Authors:
+ *   ZW, Zef Wolffs, Nikhef, zefwolffs@gmail.com
+ *
+ * Copyright (c) 2022, CERN
+ *
+ * Redistribution and use in source and binary forms,
+ * with or without modification, are permitted according to the terms
+ * listed in LICENSE (http://roofit.sourceforge.net/license.txt)
+ */
+
+#include "RooFit/MultiProcess/HeatmapAnalyzer.h"
+
+#include "TSystemDirectory.h"
+#include "TList.h"
+
+#include <fstream>
+
+namespace RooFit {
+namespace MultiProcess {
+
+/** \class HeatmapAnalyzer
+ *
+ * \brief Reads and processes logfiles produced by RooFit::MultiProcess::ProcessTimer
+ *
+ * RooFit::MultiProcess::ProcessTimer records timings of multiple processes simultaneously
+ * and allows for these timings to be written out in json format, one for each process.
+ * This class, the HeatmapAnalyzer, can read these json files and produce a heatmap from
+ * them with partial derivatives on the y-axis, likelihood evaluations on the x-axis, and
+ * time expenditures on the z-axis. This class also contains some convenience functions
```

Comment:
What's the unit of the time expenditure on the z-axis? Can you write this here pls?

Maybe you can also add a Z-axis label to the histogram to make that clear also to people that are too lazy to read the docs.

Reply:
Time expenditure is in the units it was logged in on the ProcessTimer, so I think it does not make sense I think to mention here the units. The HeatmapAnalyzer does not know anything about units but just sees numbers from the log files. I will change the ProcessTimer to make it clear what units timings are logged in, is that alright?

Reply:
Yes, that's great! Thanks for the clarification